apis/s3/src/main/java/org/jclouds/s3/filters/AwsHostNameUtils.java

```diff
@@ -26,7 +26,7 @@
 
 public class AwsHostNameUtils {
 
-   private static final Pattern S3_ENDPOINT_PATTERN = Pattern.compile("^(?:.+\\.)?s3[.-]([a-z0-9-]+)$");
+   private static final Pattern S3_ENDPOINT_PATTERN = Pattern.compile("^(?:.+\\.)?s3[.\\-]([a-z0-9-]+)(?>\\.[a-z0-9-]+)*$");
```

Comment:
Can you add some kind of unit test which exercises this?

Reply:
Done.